app/models/service.rb

```diff
@@ -181,29 +181,35 @@ def shutdown_guest
   end
 
   def power_states_match?(action)
-    if composite? && (power_states.uniq == map_composite_power_states(action))
-      return update_power_status(action)
-    elsif atomic? && (power_states[0] == POWER_STATE_MAP[action])
-      return update_power_status(action)
-    end
+    return update_power_status(action) if all_states_match?(action)
```

Comment:
Could simplify to: `all_states_match?(action) ? update_power_status(action) : false`